File: solutions/soda_slurper/soda_slurper.py
def main() -> None:
    e, f, c, *_ = tuple(map(int, input().split()))

    count = e + f
    price = c

    drunk = 0

    while count >= price:
        cbp: int = count // price

        count = count % price
        count += cbp
        drunk += cbp

    print(drunk)


if __name__ == '__main__':
    main()

File: solutions/soda_slurper/soda-slurper.cpp
#include<iostream>

int main() {
    unsigned int e, f, c;
    std::cin >> e >> f >> c;

    unsigned int count = e + f;
    const unsigned int price = c;

    unsigned int drunk = 0;

    while (count >= price) {
        unsigned int cbp = count / price;

        count = count % price;
        count += cbp;
        drunk += cbp;
    }

    std::cout << drunk << std::endl;
}
